libsolidity/ast/ASTJsonExporter.cpp

```diff
@@ -468,6 +468,10 @@ bool ASTJsonExporter::visit(FunctionDefinition const& _node)
 		attributes.emplace_back("functionSelector", _node.externalIdentifierHex());
 	if (!_node.annotation().baseFunctions.empty())
 		attributes.emplace_back(make_pair("baseFunctions", getContainerIds(_node.annotation().baseFunctions, true)));
+
+	if (_node.annotation().internalFunctionID.set())
```

Comment:
This would need to both check that the ``SetOnce`` is set *and* that the ``optional`` has a value, wouldn't it? I.e.
```suggestion
	if (_node.annotation().internalFunctionID.set() && _node.annotation().internalFunctionID->has_value())
```

Reply:
Yeah, looks like it's the latter - so we can just drop the `optional` part.

Reply:
Yeah, [set()](https://github.com/ethereum/solidity/blob/960889f532aad8f5de0fbb3bf0822f349300fb5f/libsolutil/SetOnce.h#L83) already check if it `has_value()`. Maybe we should rename it to `is_set()` or something.